builds.sh

```diff
@@ -130,11 +141,18 @@ for sourceId; do
 				end
 			')"
 
+			if [ -n "${imageArchResolved["$lookup"]:+x}" ]; then
```

Comment:
You've kept the `-n`, but swapped the bodies of the `if` and the `else`, which is the same as using `-z` and not swapping the bodies.